adapters/final_characters_controller_test.go

```diff
@@ -0,0 +1,205 @@
+package adapters
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"testing"
+
+	"github.com/SergioSanchezWizeline/academy-go-q12021/domain/model"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/interfaces"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/mocks"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/usecase"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestNewFinalCharactersController(t *testing.T) {
+	type args struct {
+		usecase usecase.FinalCharactersUserCase
+	}
+	tests := []struct {
+		name string
+		args args
+		want interfaces.Controller
+	}{
+		{
+
+			"New controller success",
+			args{&mocks.FinalCharactersUserCase{}},
+			&finalCharactersController{&mocks.FinalCharactersUserCase{}},
```

Comment:
```suggestion
			name: "New controller success",
			args: args{&mocks.FinalCharactersUserCase{}},
			want: &finalCharactersController{&mocks.FinalCharactersUserCase{}},
```

It's a good practice to leave the name of the fields when creating testing objects. Right now it's very readable, but with time, it can be very difficult to track the values of each test.

Please apply this to all the tests that are lacking it.

Reply:
Done